docs/source/posts/2022/2022-08-09-week-8-mohamed.rst

```diff
@@ -0,0 +1,37 @@
+Week 8: Back to the shader-based version of the `Timeline`
+==========================================================
+
+.. post:: August 9 2022
+   :author: Mohamed Abouagour
+   :tags: google
+   :category: gsoc
+
+
+What did you do this week?
+--------------------------
+
+- First, I made some modifications to the main animation PR. Then as Filipi requested, I integrated the vertex shader That I was implementing a few weeks ago to work with the new improved version of the ``Timeline``.
+
+- As for how keyframes are sent to the GPU, the method being used is to send the needed keyframes for each draw. This is heavy because we only roll out the interpolation part, which with linear or step interpolation won't make any difference! Instead, I tested hard-setting all the keyframes as a constant variable by manipulating the shader program's string. This also was slow to initialize, and the shader was getting bigger and slower to bind and unbind. To solve this problem, I made a uniform that holds all the keyframes of the animation and sent data as vectors with was faster than string manipulation, also it was faster to render since data are now stored directly in memory and the shader program was a lot more compact. But this method had an issue; uniforms do not keep data stored as expected! If two or more actors have the same uniform name in their shader program. And only one of them was set, the other actor will get this value as well. A way around this is to change the names of the uniforms so that they maintain their data.
```

Comment:
maybe `, which` instead of `with`, in the line ` I made a uniform that holds all the keyframes of the animation and sent data as vectors with was faster than string manipulation`